### PR TITLE
add functions for quantileExactLow & quantileExactHigh

### LINKED ISSUE #13742: quantileExactLow, quantileExactHight
Users want to get rounding mode in `quantileExact` equivalent to "median_low" from Python:
https://docs.python.org/3/library/statistics.html#statistics.median_low
while current implementation works equivalent to "median_high" (?)

### PR DESCRIPTION
**Pull request by bharatnc**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):

- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add `quantileExactLow` and `quantileExactHigh` implementations with respective aliases for `medianExactLow` and `medianExactHigh`.

Detailed description / Documentation draft:

This PR adds the quantileExactLow and quantileExactHigh
implementations which are equivalent to how the `median_low`
and `median_high` functions are implemented in python.

`median_low`: https://github.com/python/cpython/blob/master/Lib/statistics.py#L438
`median_high`: https://github.com/python/cpython/blob/master/Lib/statistics.py#L460

The `exactQuantileHigh` implementation returns exact values as the `exactQuantile`. Also `exactMedianLow` and `exactMedianHigh` are being added as aliases for `exactQuantileLow` and `exactQuantileHigh` respectively.

If the value of the level is 0.5 then median is calculated on the sorted array - otherwise the element at the index corresponding to the value of `level * size_array` is returned.

Closes #13742

### CODE REVIEW

**Comment by bharatnc** [comment]
This PR addresses #13818.

File: docs/en/sql-reference/aggregate-functions/reference/quantileexact.md [hunk 1]
```diff
@@ -49,6 +49,114 @@ Result:
...
+```
+
+Result:
+
+``` text
+┌─quantileExactLow(number)─┐
+│                        4 │
+└──────────────────────────┘
+```
+# quantileExactHigh {#quantileexactHigh}
```
Comment (bharatnc) [inline docs/en/sql-reference/aggregate-functions/reference/quantileexact.md:110]: Waiting until builds finish before fixing - one failed doc check as this link should be `{#quantileexacthigh}` - `H` not capital.

Comment (bharatnc) [inline docs/en/sql-reference/aggregate-functions/reference/quantileexact.md:110]: - [ ] fix `{#quantileexactHigh}` -> `{#quantileexacthigh}` (fails doc check)

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by bharatnc** [comment]
Thank you @nikitamikhaylov !

**Pull request merged**